src/values.rs

```diff
@@ -389,14 +405,17 @@ impl Value {
                             let key = key.to_bytes_le();
                             let value = value.to_ptr(arena, registry, &info.ty)?;
 
-                            let value_malloc_ptr = libc_malloc(elem_layout.size());
+                            let index = value_map.mappings.len();
```

Comment:
Here we could also use `enumerated` instead of calculating the length in each iteration, right?